dev/planning-example.py

```diff
@@ -0,0 +1,292 @@
+
+# longitude and latitude would be represented internally by astropy
+# angles.Longitude/Latitude.  So any value that can naturally initialize
+# one of those objects could be passed as the parameter to Observer
+
+# It's also possible to initialize using astropy objects and other
+# direct entities
+from astropy.coordinates.angles import Longitude, Latitude
+
+obs = Observer(name='Keck 1',
+               longitude=Longitude(155.4750 * u.degree),
+               latitude=Latitude(19.8264 * u.degree),
+               elevation=4160 * u.m,
+               pressure=0.615 * u.bar,
+               relative_humidity=0.11,
+               temperature=0.0 * u.deg_C,
+               timezone=pytz.timezone('US/Hawaii'),
+               description="W.M. Keck 1 Telescope on Mauna Kea, Hawaii")
+
+# It would also be desirable to be able to have a small database of
+# common telescopes.  Maybe this can at first simply take the form of
+# a python module:
+from astroplan import sites
+
+obs = sites.Keck1
+
+# Environmental conditions should be updatable.  Suggest using keyword
+# parameters (this would make it easy to update an internal AltAz coordinate):
+
+obs.set_environment(pressure=0.600 * u.bar, relative_humidity=0.2,
+                    temperature=10 * u.deg_C)
+
+# It's easy to construct date/times from the observer, since it knows about
+# its timezone.  These dates can be passed directly to target check methods.
+
+# dates default to midnight if not otherwise specified
+# Date can be anything that astropy.time.Time will accept
+day = obs.get_date("2015-05-01")
+
+# Can get more and more specific
+t1 = obs.get_date("2015-05-01 21:00")
+t1 = obs.get_date("2015-05-01 21:33:41")
+
+# default is current time
+cur_time = obs.get_date()
+
+# Dates are assumed to be in the observer's time zone unless a time zone
+# specifier is given
+t2_utc = obs.get_date("2015-05-01 21:33:41 UTC")
+
+"""
+We can ask about typical times of interest at this observing position
+Returned dates are assumed to be in the observer's time zone for convenience.
+The date parameter, if given, can be an astropy Time or a datetime instance.
+If no parameter is given, the current date/time is assumed.
+"""
+
+# sunset
+obs.next_sunset(date=day)
+obs.previous_sunset(date=day)
+
+# sunrise
+obs.next_sunrise(date=day)
+obs.previous_sunrise(date=day)
+
+# moon rise
+obs.next_moon_rise(date=day)
+obs.previous_moon_rise(date=day)
+
+# moon set
+obs.next_moon_set(date=day)
+obs.previous_moon_set(date=day)
+
+# The above functions can be called with an `angle` keyword argument to specify
+# a particular horizon angle for rising or setting, or can be called with
+# convenience functions for particular morning/evening twilight:
+
+obs.next_sunset(date=day, angle=18*u.degree) # (i.e., astronomical twilight)
+
+# evening (astronomical) twilight
+obs.evening_astronomical(date=day)
+
+# evening (nautical) twilight
+obs.evening_nautical(date=day)
```

Comment:
I think yes, we need "next" and "previous" for all those functions that take a `date`.
If it's ambiguous whether `sunset` is the previous or next one, it's also ambiguous for `evening`, no?

Is a keyword argument called `which` or ??? that can be set to 'previous', 'next' and maybe 'closest' better that separate methods? It does keep the API smaller. The question there is which is the best default and whether users would be bitten often by forgetting to set the option or not being aware of it at all, and getting something other than they expect?